### PR TITLE
Make eventuallyQueue initialization not rely on ParsePlugins

### PR DESCRIPTION
**Pull request by wangmengyan95**
When we initialize Parse, we initialize eventuallyQueue async. Thus, sometimes in our integration test, when we actually initialize eventuallyQueue, the `ParsePlugins` is not available. This will cause a NPE. This change make eventuallyQueue initialization not rely on `ParsePlugins`, we pass a context object to the initialization function instead of getting it from `ParsePlugins`.

### CODE REVIEW

File: Parse/src/main/java/com/parse/Parse.java [hunk 1]
```diff
@@ -600,6 +601,10 @@ static void checkCacheApplicationId() {
    */
   /* package */ static ParseEventuallyQueue getEventuallyQueue() {
     Context context = ParsePlugins.Android.get().applicationContext();
+    return getEventuallyQueue(context);
+  }
+
+  /* package */ static ParseEventuallyQueue getEventuallyQueue(Context context) {
```
Comment (grantland) [inline Parse/src/main/java/com/parse/Parse.java]: We can make this `private` for now

File: Parse/src/main/java/com/parse/Parse.java [hunk 2]
```diff
@@ -369,7 +370,7 @@ public static void initialize(Context context, String applicationId, String clie
     );
   }
 
-  public static void initialize(Configuration configuration) {
+  public static void initialize(final Configuration configuration) {
```
Comment (grantland) [inline Parse/src/main/java/com/parse/Parse.java]: let's just hold a `final` reference of `Context` instead of the whole `configuration`

Comment (wangmengyan95) [inline Parse/src/main/java/com/parse/Parse.java]: kk

File: Parse/src/main/java/com/parse/Parse.java (hunk 1 above)
Comment (wangmengyan95) [inline Parse/src/main/java/com/parse/Parse.java]: :+1:

File: Parse/src/main/java/com/parse/Parse.java [hunk 3]
```diff
@@ -402,13 +403,13 @@ public static void initialize(Configuration configuration) {
     // Make sure the data on disk for Parse is for the current
     // application.
     checkCacheApplicationId();
-    new Thread("Parse.initialize Disk Check & Starting Command Cache") {
+    Task.callInBackground(new Callable<Void>() {
```
Comment (grantland) [inline Parse/src/main/java/com/parse/Parse.java]: can we `Task.call(() -> { ... }, Task.BACKGROUND_EXECUTOR)` instead?

**Comment by grantland** [comment]
LGTM

**Pull request merged**